src/spartan_handler_fsm.erl

```diff
@@ -244,54 +217,55 @@ reply_fail(_State1 = #state{dns_message = DNSMessage, from = {FromModule, FromKe
 
 %% @private
 resolve(Parent, Upstream, State) ->
-    try do_resolve(Parent, Upstream, State) of
-        _ ->
-            ok
+    try
+        do_resolve(Parent, Upstream, State)
     catch
-        exit:Exception ->
-            lager:warning("Resolver (~p) Process exited: ~p stacktrace ~p", [Upstream, Exception, erlang:get_stacktrace()]),
-            %% Reraise the exception
-            exit(Exception);
-        Reason ->
-            lager:warning("Resolver (~p) Process exited: ~p stacktrace ~p", [Upstream, Reason, erlang:get_stacktrace()]),
-            exit(unknown_error)
+        Class:Reason ->
+            StackTrace = erlang:get_stacktrace(),
+            lager:warning("Resolver (~p) Process exited: ~p stacktrace ~p",
+                [Upstream, Reason, erlang:get_stacktrace()]),
+            erlang:raise(Class, Reason, StackTrace)
     end.
 
 do_resolve(Parent, Upstream = {UpstreamIP, UpstreamPort}, #state{data = Data, from = {spartan_udp_server, _}}) ->
-    lager:debug("Sending query to Upstream: ~p", [Upstream]),
```

Comment:
I propose to use `recon_trace` for this purpose. It should be much simpler for us and for customers to run one command instead of changing spartan's configuration files, restarting service, resending logs and so on.

```
/opt/mesosphere/packages/spartan*/spartan/bin/spartan-env eval "recon_trace:calls({spartan_handler_fsm, do_resolve, '_'}, 10, [{scope, local}]), timer:sleep(10000)."
```